### DISCUSSION TITLE
HEIMAN HS2WD-E use case as a Homekit Siren

### DISCUSSION THREAD

**Discussion by remussamoila** [Koenkk/zigbee2mqtt#17924] [Devices]
I am testing this HEIMAN HS2WD-E siren, that is recognized by Zigbee2MQTT flawlessly.
Unfortunately, the homekit app shows that the Siren is incompatible, showing the battery level and charging state.
In order to use the device, I configured a plugin by [arachnetech] (https://github.com/arachnetech/homebridge-mqttthing) with two switches for on and off triggering and it is fairly working with this kind of messages
- starting 
 "setOn": {
                    "topic": "zigbee2mqtt/Siren/set",
                    "apply": "return JSON.stringify({\"warning\": { \"mode\": \"emergency\", \"level\": \"high\", \"strobe_level\": \"high\", \"strobe\": true, \"strobe_duty_cycle\": 10, \"duration\": 5 }})"
- stopping
                "setOn": {
                    "topic": "zigbee2mqtt/Siren/set",
                    "apply": "return JSON.stringify({\"warning\": { \"mode\": \"stop\", \"strobe\": false, \"duration\": 1 }})"
Is there something that could be improved to recognize this device properly on homekit without any workaround with messages?
Thanks!

**Comment by Jk607617627** [comment]
Hello, is there something new in this topic? I have bought the siren today and have the same issue as the author of this topic.
Thank you

**Reply by remussamoila** [reply]
Lucaese solution is on the link below. I test it and it's working for me too, using this plugin: https://github.com/arachnetech/homebridge-mqttthing#readme

https://github.com/arachnetech/homebridge-mqttthing/issues/615

My adapted code for the Mqttthing plugin:
{
            "type": "switch",
            "name": "Indoor Siren",
            "topics": {
                "setOn": "zigbee2mqtt/Siren/set"
            },
            "onValue": "{\"warning\":{\"duration\":1800,\"level\":\"very_high\",\"mode\":\"burglar\",\"strobe\":true,\"strobe_duty_cycle\":10,\"strobe_level\":\"low\"}}",
            "offValue": "{\"warning\":{\"duration\":60,\"level\":\"low\",\"mode\":\"stop\",\"strobe\":false,\"strobe_duty_cycle\":1,\"strobe_level\":\"low\"}}",
            "history": false,
            "manufacturer": "Heiman",
            "model": "HS2WD-E",
            "accessory": "mqttthing"
        }

I now have a homekit switch to start or stop the siren.
